staging/src/k8s.io/client-go/tools/record/events_cache.go

```diff
@@ -499,7 +500,7 @@ func (c *EventCorrelator) EventCorrelate(newEvent *v1.Event) (*EventCorrelateRes
 	}
 	aggregateEvent, ckey := c.aggregator.EventAggregate(newEvent)
 	observedEvent, patch, err := c.logger.eventObserve(aggregateEvent, ckey)
-	if c.filterFunc(observedEvent) {
+	if c.filterFunc(observedEvent) && newEvent.Reason != events.ContainerHealthy{
```

Comment:
the event recorder is used by many other subsystems which may have similar reasons for omitting how/when particular events are filtered.  rather than couple the events_cache to know about a particular type of event reason for kubelet sourced events, we should structure this a little different so each source component could provide its filter variance appropriate for the component.

For example, in kubelet server.go `NewMainKubelet` when we create the EventRecorder, we could use the new event recorder with correlator options, and augment correlator options to add an additional fillter func specific to kubelet?

Reply:
In EventCorrelate function for filtering events，Add a health function to the filter function to determine that the reason is healthy, and release the flow restriction on healthy events.